cartographer/ground_truth/autogenerate_ground_truth_main.cc

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cmath>
+#include <fstream>
+#include <string>
+
+#include "cartographer/common/port.h"
+#include "cartographer/ground_truth/proto/relations.pb.h"
+#include "cartographer/mapping/proto/sparse_pose_graph.pb.h"
+#include "cartographer/transform/transform.h"
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+
+DEFINE_string(
+    pose_graph_filename, "",
+    "File with the pose graph proto from which to generate ground truth data.");
+DEFINE_string(output_filename, "", "File to write the ground truth proto to.");
+DEFINE_double(min_covered_distance, 100.,
+              "Minimum distance covered before a loop closure is considered a "
+              "candidate for autogenerated ground truth.");
+DEFINE_double(outlier_threshold, 0.15,
+              "Distance in meters beyond which constraints are considered "
+              "outliers.");
+
+namespace cartographer {
+namespace ground_truth {
+namespace {
+
+std::vector<double> ComputeCoveredDistance(
```

Comment:
Seems redundant to me. `ComputeCoveredDistanceFromStart()` maybe? I'd favor conciseness, but let me know if you would like me to add `FromStart`.